lib/auth/machineid/workloadidentityv1/expression/evaluate_test.go

```diff
@@ -0,0 +1,88 @@
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package expression_test
+
+import (
+	"testing"
+
+	traitv1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/trait/v1"
+	workloadidentityv1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/workloadidentity/v1"
+	"github.com/gravitational/teleport/lib/auth/machineid/workloadidentityv1/expression"
+	"github.com/stretchr/testify/require"
+)
+
+func TestEvaluate(t *testing.T) {
```

Comment:
It'd be nice to codify some of the additional edge-cases here so we have reference for how this behaves (e.g nil struct, unknown map key) etc.